python/services/github_service.py

```diff
@@ -65,6 +67,34 @@ def __init__(self, org_token: str, organisation_name: str) -> None:
         ), execute_timeout=60)
         self.organisation_name: str = organisation_name
 
+    def archive_inactive_repositories(self, repo_type_to_archive: str, last_active_cutoff_date: datetime,
+                                      allow_list: list[str]) -> None:
+        for repo in self.__get_repos_to_consider_for_archiving(repo_type_to_archive):
+            if self.__is_repo_ready_for_archiving(repo, last_active_cutoff_date, allow_list):
+                repo.edit(archived=True)
+
+    def __get_repos_to_consider_for_archiving(self, repository_type: str) -> list[Repository]:
+        repositories = list(
+            self.github_client_core_api.get_organization(self.organisation_name).get_repos(type=repository_type))
+        return [repository for repository in repositories if not (repository.archived or repository.fork)]
+
+    def __is_repo_ready_for_archiving(self, repository, last_active_cutoff_date, allow_list: list[str]) -> bool:
```

Comment:
Just for awareness, this is the function I considered moving into a strategy pattern 👀

It contains the meat of our archiving Policy, and it would be nice to have the logic explicitly defined in a separate Class that we inject into services

But that seemed like a little much 😂